proto/ethermint/evm/v1alpha1/query.proto

```diff
@@ -240,3 +246,22 @@ message QueryStaticCallRequest {
 message QueryStaticCallResponse {
   bytes data = 1;
 }
+
+// CallArgs represents the arguments for a eth_call.
+message CallArgs {
```

Comment:
wondering if you can use a `MsgEthereumTx` instead, seems that the fields are all shared?

Reply:
The only thing is `MsgEthereumTx.AsMessage` take a `signer` as parameter.

Reply:
In the go-ethereum, the data goes from `CallArgs` to `Message` directly, here we just created a protobuf version of `CallArgs`, use `MsgEthereumTx` would make it less clean here.

Reply:
can we use a `proto.Any` instead? Otherwise, we will have to add additional fields every time we upgrade to a new geth version. What do you think?

Reply:
With `proto.Any` do we need to serialize it to bytes? Or we just pass the json as a string? after all we are trying to keep it compatible with geth parameters.

Reply:
I changed to use json string, copied the whole `CallArgs` implementation from go-ethereum, so it'll be compatible with geth.

Reply:
> With proto.Any do we need to serialize it to bytes? Or we just pass the json as a string? after all we are trying to keep it compatible with geth parameters.

Because we are only using the `EthCall` query internally it's ok to use `TxData`. Then we can also validate the fields and support multiple tx types.

Reply:
summarizing:

- use ` Call(args rpctypes.CallArgs, ... )`
- create a `TxData` from args
- `EthCallRequest` should contain the `TxData` and `From` fields
- create a `ethtypes.NewMessage` from the `TxData` and `From` fields

Reply:
what do you think about current json string approach, we can reuse `CallArgs` and `CallArgs.AsMessage` which is copied from go-ethereum directly, easier to keep if compatible with geth, don't need to convert data by ourself.

Reply:
`CallArgs` is being deprecated in the next version (v1.10.4) in favor of `TransactionArgs`.

I also don't like the approach of using json as the request field over gRPC tbh. The only place I've seen it is on IBC for ACKs and `PacketData` but just because chains can have different encoding formats, and here we don't fall under that case.

Let's maybe add the validation and then move the format to a new [discussion](https://github.com/tharsis/ethermint/discussions)

Reply:
It would be ideal if we can have a `CallArgs` defined in protobuf while has the compatible json representation with geth, not sure if that's possible.

Reply:
> Let's maybe add the validation and then move the format convo to a new [discussion](https://github.com/tharsis/ethermint/discussions)

I think there's no extra validation on args in geth: https://github.com/ethereum/go-ethereum/blob/v1.10.3/internal/ethapi/api.go#L881.
The basic validation is already done in json unmarshal.

Reply:
> `CallArgs` is being deprecated in the next version (v1.10.4) in favor of `TransactionArgs`.

I guess we can copy the `TransactionArgs` then ;D

Reply:
ideally `TransactionArgs` should be made public in geth... cc: @s1na 🙏